README.md

```diff
@@ -1,981 +1,264 @@
 
-* The maximum line length allowed is 100 characters instead of 80 characters
+* [Go modules](https://go.dev/ref/mod)
 
-In addition to `flake8`, docstrings are also enforced by the plugin `flake8-docstrings` with
-the following exceptions:
+## Goals
 
-* D100: Missing docstring in public module
-* D104: Missing docstring in public package
-* D105: Missing docstring in magic method
+Please note that Cachi2 is rather picky, aiming to:
 
-The format of the docstrings should be in the
-[reStructuredText](https://docs.python-guide.org/writing/documentation/#restructuredtext-ref) style
-such as:
+* encourage or enforce best practices
+* enforce building from source - no pre-built artifacts, such as Python [wheels][wheel-spec]
+* never execute arbitrary code - looking at you [setup.py (discouraged)][setuppy-discouraged]
+* keep the implementation simple
 
-```python
-Set the state of the request using the Cachito API.
+To play nicely with Cachi2, the build process for your project must be
 
-:param int request_id: the ID of the Cachito request
-:param str state: the state to set the Cachito request to
-:param str state_reason: the state reason to set the Cachito request to
-:return: the updated request
-:rtype: dict
-:raise CachitoError: if the request to the Cachito API fails
-```
+* **Defined** - Cachi2 only fetches dependencies that are explicitly declared - typically in a lockfile generated by
+  your package manager.
+* **Reproducible** - Cachi2 will refuse to fetch a dependency if it's not pinned to an exact version. This goes
+  for transitive dependencies as well (and ties to the Defined point). Most package managers pin all dependencies
+  automatically in lockfiles.
+* **Secure** - Even with a lockfile, your build is not truly safe from supply chain attacks unless you verify the
+  checksums of all dependencies. If your package manager supports specifying the expected checksums, we strongly
+  encourage you to make use of them.
 
-Additionally, `black` is used to enforce other coding standards.
+  ⚠ Cachi2 will verify checksums if present, but doesn't require them by default. This may change in the future.
 
-To verify that your code meets these standards, you may run `tox -e black,flake8`.
+In return, Cachi2 will help make your build
 
-## Quick Start
+* **Auditable** - by generating a manifest of all the dependencies that go into your build.
 
-Run the application locally (requires [docker-compose](https://docs.docker.com/compose/)):
+The ability to achieve the goals depends on the hermeticity of the build process. Ideally, you should try to isolate the
+build from both the internet and the underlying host system to avoid implicit dependencies, irreproducible behavior and
+whole hosts of other issues. Cachi2 itself is not a hermetic build system. We suggest you take advantage of existing
+technologies - such as containers - to achieve isolation (see [usage](#usage)).
 
-```bash
-make run
-```
+## Usage
```

Comment:
It's just a suggestion, but I was thinking we could keep a very brief usage section here in the main README (just an example on how to invoke the CLI), and move this full example to a separate doc. 

I tend to like smaller, more focused docs instead of longer ones, but that's probably just me nitpicking :innocent:

Reply:
Yeah that's a good point. I'll see how it would work

Reply:
Moved to separate doc. I do prefer it this way :+1: